arquillian-chameleon-deployments/arquillian-chameleon-maven-build-deployment/src/main/java/org/arquillian/container/chameleon/deployment/maven/MavenBuildAutomaticDeployment.java

```diff
@@ -49,9 +52,35 @@
             configurationDistributionStage.addProperty(properties[i], properties[i + 1]);
         }
 
-        return configurationDistributionStage
+        BuiltProject build = configurationDistributionStage
             .ignoreFailure()
-            .build().getDefaultBuiltArchive();
+            .build();
+
+        if (isModuleSet(conf)) {
+            final File projectDirectory = build.getModel().getProjectDirectory();
+            final Path normalize = projectDirectory.toPath().normalize();
+
+            final String relativeModuleDirectory = conf.module().replace('/', File.separatorChar);
+
+            build = getSubmodule(build,
+                normalize.toString() + File.separator + relativeModuleDirectory + File.separator + "pom.xml");
+        }
+
+        return build.getDefaultBuiltArchive();
+
+    }
+
+    private boolean isModuleSet(MavenBuild conf) {
+        return !"".equals(conf.module());
```

Comment:
I would check here for non-empty string. Because this should fail if user set module to `""` by mistakenly.

Reply:
No it is not going to fail, it is going to take the current module https://github.com/arquillian/arquillian-container-chameleon/pull/89/files#diff-9885e923debaf99aedf27078628f4a3cR69